clcache.py

```diff
@@ -74,6 +80,80 @@ def __str__(self):
         return repr(self.message)
 
 
+class ManifestsManager(object):
+    HASH_INVALIDATION_COUNTER = 3
+
+    def __init__(self, manifestsRootDir):
+        self._manifestsRootDir = manifestsRootDir
+
+    def manifestDir(self, manifestHash):
+        return os.path.join(self._manifestsRootDir, manifestHash[:2])
+
+    def manifestPath(self, manifestHash):
+        return os.path.join(self.manifestDir(manifestHash), manifestHash + ".json")
+
+    def setManifest(self, manifestHash, manifest):
+        ensureDirectoryExists(self.manifestDir(manifestHash))
+        with open(self.manifestPath(manifestHash), 'w') as outFile:
+            # Convert namedtupel to dict to get key names
+            doc = {
+                'includeFiles': manifest.includeFiles,
+                'includesContentToObjectMap': manifest.includesContentToObjectMap
+            }
+            json.dump(doc, outFile, sort_keys=True, indent=2)
```

Comment:
Very nice. Namedtuple -> _asdict() -> OrderedDict -> JSON preserves the key names as well as the order.